plumbing/msg/sender.go

```diff
@@ -54,12 +58,13 @@ type Sender struct {
 
 // NewSender returns a new Sender. There should be exactly one of these per node because
 // sending locks to reduce nonce collisions.
-func NewSender(signer types.Signer, chainReader chain.ReadStore, blockTimer core.BlockTimer,
+func NewSender(signer types.Signer, chainReader chain.ReadStore, cst *hamt.CborIpldStore, blockTimer core.BlockTimer,
```

Comment:
An alternate proposal for a cleanup followon: tipsets are given a pointer to a state tree upon creation and `ts.State()` returns the tree.  This should be evaluated in more depth before going down this road

Reply:
I'll add an issue to clean this up after this PR is merged.

Fwiw, I would prefer to add a `GetLatestState` method or function somewhere, rather than adding more state to the TipSet struct. I think linking low level objects to their containers is a bad pattern as it introduces a lot of undefined behavior when dealing with those objects in isolation.

For example, we may want to temporarily create isolated TipSets for tests or certain chain operations, and leaving the store pointer nil could introduce mistakes if we ever call a method that utilizes said pointer.